### PR TITLE
[Please see PR 541] Removed unneeded code from aec_sync

### PR DESCRIPTION
**Pull request by ThomasArts**
Removed unused code in aec_sync that had no further purpose.
Updated the aecore_sync_SUITE trying to provoke an error, added that test later to aec_eunit_SUITE instead.

[Finished 153772644]

### CODE REVIEW

File: apps/aecore/src/aec_sync.erl [hunk 1]
```diff
@@ -215,8 +197,7 @@ sync_worker() ->
     process_job(Res).
 
 %% Note: we always dequeue exactly ONE job
-process_job([{T, Job}]) ->
-    reg_worker(Job, T),
```
Comment (lucafavatella) [inline apps/aecore/src/aec_sync.erl]: I see, usage of this was removed in df8f0f28

File: apps/aecore/test/aecore_sync_SUITE.erl [hunk 2]
```diff
@@ -87,46 +85,27 @@ init_per_suite(Config) ->
...
+    config({devs, [dev1, dev2]}, Config);
 init_per_group(three_nodes, Config) ->
-    config(three_nodes, Config);
+    config({devs, [dev1, dev2, dev3]}, Config);
 init_per_group(one_blocked, Config) ->
-    Config1 = config(two_nodes, Config),
+    Config1 = config({devs, [dev1, dev2]}, Config),
     preblock_second(Config1),
     Config1;
+init_per_group(faulty_connect, Config) ->
```
Comment (lucafavatella) [inline apps/aecore/test/aecore_sync_SUITE.erl]: Is this meant to be deleted as moved to aec_eunit_SUITE?

Comment (ThomasArts) [reply apps/aecore/test/aecore_sync_SUITE.erl]: Yes, will remove.

File: apps/aecore/test/aecore_sync_SUITE.erl [hunk 3]
```diff
@@ -144,12 +123,35 @@ end_per_testcase(_Case, Config) ->
...
 
-test_subscription(_Config) ->
-    N = aecore_suite_utils:node_name(dev1),
+start_second_node(Config) ->
+    [ Dev1, Dev2 | _ ] = proplists:get_value(devs, Config),
+    N1 = aecore_suite_utils:node_name(Dev1),
+    N2 = aecore_suite_utils:node_name(Dev2),
+    aecore_suite_utils:start_node(Dev2, Config),
+    aecore_suite_utils:connect(N2),
+    timer:sleep(2000),
```
Comment (uwiger) [inline apps/aecore/test/aecore_sync_SUITE.erl]: Shouldn't this be changed to `aecore_suite_utils:await_aehttp(N2)`?

Comment (ThomasArts) [reply apps/aecore/test/aecore_sync_SUITE.erl]: Yes.

Comment (lucafavatella) [reply apps/aecore/test/aecore_sync_SUITE.erl]: I see two approvals. Merging this.
I am preparing separate PR for optimizing this.

Comment (lucafavatella) [reply apps/aecore/test/aecore_sync_SUITE.erl]: doh merge conflict

File: apps/aecore/test/aecore_sync_SUITE.erl [hunk 4]
```diff
@@ -168,36 +170,29 @@ test_subscription(_Config) ->
             ok
     end.
 
-mine_on_first(_Config) ->
-    N = aecore_suite_utils:node_name(dev1),
+mine_on_first(Config) ->
+    [ Dev1 | _ ] = proplists:get_value(devs, Config),
+    N = aecore_suite_utils:node_name(Dev1),
     aecore_suite_utils:mine_one_block(N),
```
Comment (lucafavatella) [inline apps/aecore/test/aecore_sync_SUITE.erl]: This was the little conflict. Solved locally, testing.

File: apps/aecore/test/aecore_sync_SUITE.erl (hunk 3 above)
Comment (lucafavatella) [reply apps/aecore/test/aecore_sync_SUITE.erl]: Testing merge on my fork https://travis-ci.org/lucafavatella/epoch/builds/320305550
Then will attempt to push as GitHub recommends (force-pushing would require new approvals).

Comment (lucafavatella) [reply apps/aecore/test/aecore_sync_SUITE.erl]: The protected branch hook does not allow me to push to master as GitHub recommends.

Comment (lucafavatella) [reply apps/aecore/test/aecore_sync_SUITE.erl]: Opened PR #541.

**Pull request closed without merging**